test/conformance/runtime/readiness_probe_test.go

```diff
@@ -152,6 +152,9 @@ func TestProbeRuntime(t *testing.T) {
 // which may not be what a user wants.
 // See https://github.com/knative/serving/issues/10765.
 func TestProbeRuntimeAfterStartup(t *testing.T) {
+	if test.ServingFlags.DisableOptionalAPI {
+		t.Skip("Container.readinessProbe is not required by Knative Serving API Specification")
```

Comment:
fwiw I do see a section about readiness probes in the Spec (https://github.com/knative/specs/blob/3dfc2e79f0f7d683004e077e76812323245c3956/specs/serving/runtime-contract.md#meta-requests), so I think this is too strong. Perhaps it's fair to say "the behaviour of readinessProbes after startup is not required by ..", although I'd suggest we really should change that!

Reply:
Agree to that. I don't think offering the API is optional but **using** the API is optional (from a user's PoV).

Reply:
Per https://github.com/knative/specs/blob/main/specs/serving/knative-api-specification-1.0.md#container, offering the API is optional.

Reply:
Am I misreading "The core.v1.Container object allows specifying both a readinessProbe and a livenessProbe"? Is that really compatible with saying "the serving specification doesn't require container.readinessProbe"? I think it does require that it be allowed, it just doesn't specify the semantics of it (unfortunately, imo, but there we are)

Reply:
(to be clear I'm ok skipping the test, I just think the wording of the skip implies readinessProbe could be  not be permitted, but in fact it is explicitly called out as permitted to be set)

Reply:
https://github.com/knative/specs/blob/main/specs/serving/knative-api-specification-1.0.md#container: Schema Requirement for `readinessCheck` is `RECOMMENDED`.

I think we need some wording fix here. @evankanderson could you take a look?

Reply:
Interesting, okay. I guess if the schema requirement isn't there, this is fine to skip as optional API. Thanks for clarifying.

Reply:
If this PR looks OK to you, could you approve it?

Reply:
I think readinessProbe is required to be allowed (by [the section I linked](https://github.com/knative/specs/blob/3dfc2e79f0f7d683004e077e76812323245c3956/specs/serving/runtime-contract.md#meta-requests)), if we want to make it optional we should clarify that section (I'd argue we should not make it optional - it breaks lots of apps if there's no way to indicate readiness other than listening on tcp).

Could we change the skip message to "Container.readinessProbe behaviour after startup is not defined by the Knative specification" since that much is unarguably correct?

Reply:
done